src/main/java/ch/ivyteam/ivy/maven/util/MavenDependencies.java

```diff
@@ -16,61 +16,51 @@
 /**
  * @since 9.2.2
  */
-public class MavenDependencies {
-  private final MavenProject project;
-  private final MavenSession session;
-  private String typeFilter;
+public interface MavenDependencies {
 
-  public MavenDependencies(MavenProject project, MavenSession session) {
-    this.project = project;
-    this.session = session;
-  }
-
-  public MavenDependencies type(String newTypeFilter) {
-    this.typeFilter = newTypeFilter;
-    return this;
-  }
-
-  public List<Path> localTransient() {
+  static List<Path> localTransient(MavenProject project) {
     return stream(project.getArtifacts())
-        .filter(this::isLocalDep)
-        .filter(this::include)
+        .filter(artifact -> isLocalDep(artifact, project))
         .map(Artifact::getFile)
         .map(File::toPath)
         .filter(Objects::nonNull)
+        .toList();
+  }
+
+  static List<Path> all(MavenProject project, MavenSession session, String typeFilter) {
```

Comment:
didn't the API look better before, less utility stile?

Reply:
both ok for me :D